main.go

```diff
@@ -211,12 +213,31 @@ func run(state overseer.State) {
 	}
 
 	ctx := context.TODO()
+	var detectorsOption engine.EngineOption
+
+	detectorCfg, err := engine.NewDetectorsConfig(*includeDetectors, *excludeDetectors)
+	if err != nil {
+		logger.Error(err, "failed to create detectors config")
+		os.Exit(1)
+	}
+	dts, err := engine.Detectors(ctx, detectorCfg)
+	if err != nil {
+		logger.Error(err, "failed to create detectors")
+		os.Exit(1)
+	}
+
+	if len(dts) > 0 {
+		detectorsOption = engine.WithDetectors(!*noVerification, dts...)
+	} else {
+		detectorsOption = engine.WithDetectors(!*noVerification, engine.DefaultDetectors()...)
+	}
```

Comment:
**Suggestion:** I think it might be easier to reason about and maintain if we have an option like `engine.WithFilterDetectors` that handles the filtering.